src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/Services/DefaultRazorSemanticTokensInfoService.cs

```diff
@@ -329,21 +329,14 @@ internal class DefaultRazorSemanticTokensInfoService : RazorSemanticTokensInfoSe
             CancellationToken cancellationToken,
             string? previousResultId = null)
         {
-            IReadOnlyList<int>? previousCSharpTokens = null;
+            var razorRanges = new List<SemanticRange>();
 
-            // Attempting to retrieve cached tokens for the C# generated document.
-            if (previousResultId != null &&
-                _csharpGeneratedDocTokensCache.TryGetValue(textDocumentIdentifier.Uri, out var documentCache) &&
-                documentCache.TryGetValue(previousResultId, out var cachedTokens))
+            if (!TryGetMinimalCSharpRange(codeDocument, out var csharpRange))
```

Comment:
Can we pass `range` in here so that we only consider the source mappings that are relevant, rather than the whole document? Or is that not what `range` is?

Reply:
This is trying to get the range that we care about for the CSharp backing document, whereas the `range` parameter for this method refers to a range inside of the Razor document. I suppose I did make it a bit ambiguous, @allisonchou feel free to give it a better name here if you'd like.

Reply:
Yeah, that's what I assumed it was, which makes it surprising that the latter isn't used to constrain the former. Seems like we're doing more work than necessary

Reply:
Actually the razor Range will be null in all cases anyway because we turned Razor-Range off due to the perf issues discussed elsewhere in this PR.